paasta_tools/contrib/get_containers_and_ips.py

```diff
@@ -0,0 +1,105 @@
+#!/usr/bin/env python
+import argparse
+import os
+import socket
+
+from paasta_tools.utils import get_docker_client
+
+HAPROXY_STATS_SOCKET = '/var/run/synapse/haproxy.sock'
+
+
+def send_to_haproxy(command):
+    s = socket.socket(socket.AF_UNIX)
+    # 1 seconds should be more than enough of a timeout since HAProxy is local
+    s.settimeout(1)
+    s.connect(HAPROXY_STATS_SOCKET)
+    print("Sending {} to HAProxy stats socket".format(command))
+    s.send((command + '\n').encode())
+    file_handle = s.makefile()
+    response = file_handle.read().splitlines()
+    print("Response: {}".format(response))
+    s.close()
+
+
+def main():
+    parser = argparse.ArgumentParser(
+        description=(
+            'Script to dump a HAProxy map between container IPs and task IDs.'
+        ),
+    )
+    parser.add_argument(
+        'map_file', nargs='?', default='/var/run/synapse/maps/ip_to_service.map',
+        help='Where to write the output map file',
+    )
+    args = parser.parse_args()
+
+    client = get_docker_client()
+
+    file_exists = os.path.isfile(args.map_file)
+    if file_exists:
+        mode = 'r+'
+    else:
+        mode = 'w'
+
+    with open(args.map_file, mode) as f:
+        if file_exists:
+            prev_lines = [line.strip().split(' ') for line in f.readlines()]
```

Comment:
Since the map files allow for any arbitrary amount of whitespace in between the key and value, it may be good to make the parsing account for that in the future (e.g. in case the map is initialized before running the script with a different amount of whitespace). This could potentially mess up the assumption that the first two keys are the key and the value. But this should be fine for our purposes.